apps/sms/views/conversation/js/conversation.js

```diff
@@ -769,10 +779,11 @@ var ConversationView = {
         return;
       }
 
+      var promises = [];
       // Recipients will exist for draft messages in threads
       // Otherwise find them from draft recipient numbers
       this.draft.recipients.forEach(function(number) {
-        Contacts.findByAddress(number).then(function(contacts) {
+        promises.push(Contacts.findByAddress(number).then(function(contacts) {
```

Comment:
and while you're here, maybe convert the functions to arrow functions, and remove the "bind" calls